source/common/common/regex.cc

```diff
@@ -47,12 +47,14 @@ class CompiledGoogleReMatcher : public CompiledMatcher {
       throw EnvoyException(regex_.error());
     }
 
-    const uint32_t max_program_size =
-        PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.google_re2(), max_program_size, 100);
-    if (static_cast<uint32_t>(regex_.ProgramSize()) > max_program_size) {
-      throw EnvoyException(fmt::format("regex '{}' RE2 program size of {} > max program size of "
-                                       "{}. Increase configured max program size if necessary.",
-                                       config.regex(), regex_.ProgramSize(), max_program_size));
+    if (config.google_re2().has_max_program_size()) {
```

Comment:
I really do think that Envoy side this check does have value. Should we consider doing the following since this field will ultimately be deleted?
1) Make a runtime key that sets the "regex program size warn level". The default value of this key would be 100.
2) Check the program size and warn if over that size? Later we could even have an error level if we want, all driven by runtime.

@ramaraochavali @htuch @jmarantz thoughts?

Reply:
+1, not all management server are capable to do the max program size check, also for handwritten config / file-based xDS. We should have a safety guard somewhere because this is a security critical config. A runtime value SGTM.

Reply:
@LisaLudique do you want to try to land some further improvements in this PR or do it in a follow up?

Reply:
For the record I agree; I just didn't go to the mat on the issue when the proposal to deprecate came.

I think there should be some baseline Envoy protection against huge regexes. But I think it might be OK to specify it differently, like as a command-line parameter rather than a per-regex override; as long as the team running the proxy has control.

Reply:
OK let's add 2 runtime keys (probably easier than CLI option), 1 for warn level and 1 for error level?

Reply:
+1 for runtime value. I assume in the "error level" we will do xDS config reject as done today?

Reply:
+1 for runtime values and a stat to indicate the same

Reply:
sgtm, will look into adding runtime keys tomorrow

Reply:
>  I assume in the "error level" we will do xDS config reject as done today?

Yes. I suggest we default the error level to 100 as it is today (unless the deprecated field is set), so we effectively get back the behavior we had before. Then we can also have a warn level which perhaps has no default (unlimited) which would issue a warn log and increment a stat.

Reply:
I pushed some commits to add the two runtime keys - please take a look and see if there are any inconsistencies with what you all described. This resulted in having to move the runtime protos into their own folder to avoid a dependency loop, and I had to update the v2 protos to make the type whisperer script work (I didn't want to special case the runtime protos).

I still need to work on adding a stat and making sure relevant docs are updated, but I wanted to ask how the file `api/envoy/type/matcher/regex.proto` is used since there is no version number.